### PR TITLE
Fix TSAN test failures under CMake

### PR DESCRIPTION
**Pull request by wilhuff**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -54,14 +54,15 @@ enable_language(CXX)
...
 list(INSERT CMAKE_MODULE_PATH 0 ${PROJECT_SOURCE_DIR}/cmake)
-include(cc_rules)
 include(compiler_id)
 include(external_rules)
 include(podspec_rules)
 include(archive_options)
 include(sanitizer_options)
 include(fuzzing_options)
 
+include(cc_rules)
```
Comment (var-const) [inline CMakeLists.txt]: Perhaps add a comment to note that `cc_rules` has to be included after `compiler_id`?

File: cmake/FindASANDylib.cmake [hunk 2]
```diff
@@ -14,8 +14,21 @@
 
 # Finds `libclang_rt.asan_osx_dynamic.dylib` on Apple platform, it is typically
 # under ${CMAKE_CXX_COMPILER}/../lib.
```
Comment (var-const) [inline cmake/FindASANDylib.cmake]: Can you expand the comment to explain why finding the `dylib` file is necessary?

File: CMakeLists.txt (hunk 1 above)
Comment (wilhuff) [reply CMakeLists.txt]: Done. And for consistency moved all the rules down here, clearly separating options processing from rule definition.

File: cmake/FindASANDylib.cmake (hunk 2 above)
Comment (wilhuff) [reply cmake/FindASANDylib.cmake]: Done.

**Pull request merged**